README.md

```diff
@@ -9,3 +9,664 @@ ExpressRouterAdapter is available as a package on NPM for use with a module bund
+
+```js
+import 'reflect-metadata'; // polyfill for aurelia-dependency-injection
+import express from 'express';
+import { ExpressRouterAdapter, RouterMetaBuilder, RouteProvider } from '@symbiotic/express-router-adapter';
+import { Container } from 'aurelia-dependency-injection';
+
+const container = new Container();
+
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .get(({ petId }) => {
+          return { petId, name: 'Sawyer '};
+        })
+    ];
+  }
+}
+container.registerAlias(PetRouter, RouteProvider);
+
+const app = express();
+container.get(ExpressRouterAdapter).applyRoutes(app);
+app.listen(4000);
+
+```
+
+### What if I have more than one router?
+
+We expose a `CompositeRouteProvider` to save you from building the composite yourself.
+
+```js
+import { CompositeRouteProvider, RouteProvider } from '@symbiotic/express-router-adapter';
+import { Container } from 'aurelia-dependency-injection';
+import { Router1, Router2 } from '.';
+
+class AppRouter extends CompositeRouteProvider {
+  static inject = [Router1, Router2];
+};
+container.registerAlias(AppRouter, RouteProvider);
+```
+
+## How do I get access to a query param?
+
+```js
+
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .query('petName')
+        .get(({pedId, petName }) => {
+          // ...
+        })
+    ];
+  }
+}
+```
+
+You can always get to the raw express request if you need to as well. And then you don't need to use the `query` method on `RouterMetaBuilder`.
+
+```js
+
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .get(({pedId, req }) => {
+          // req.query.petType
+        })
+    ];
+  }
+}
+```
+
+## What if my route handler doesn't return anything
+```js
+
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .put(({ pedId }) => {
+          // return nothing
+        })
+    ];
+  }
+}
+```
+
+Then you will automatically get a 204.
+
+
+## What if it does return something?
+
+```js
+
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .put(({ petId }) => {
+          return { petId, name: 'honey' }
+        })
+    ];
+  }
+}
+```
+
+
+Then you will automatically get a 200 with your model serialized as JSON and a content-type header of application/json.
+
+## What if I need do something asynchronous
+
+```js
+
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .put(async ({ petId }) => {
+          await doSomethingAsync(petId);
+        })
+    ];
+  }
+}
+```
+
```

Comment:
Missing a note that ERA supports async? maybe just add a comment to the put handler to call attention to the async/await support?

Reply:
No change needed